Mysterious constructor overload starring a type class

Maybe it's the late hour, but this one is so baffling it's funny:
## reproduction steps

using Scala 2.13.4,

```scala
	import reflect.ClassTag
	trait TypeClass[T]
	class Hehe[T: TypeClass](private val cls: Class[T], val tpe: Int) {
		def this(tpe: Int)(implicit hehe: TypeClass[T], tag: ClassTag[T]) =
			this(tag.runtimeClass.asInstanceOf[Class[T]], tpe)
	}
```

## problem
```
too many arguments (found 3, expected 1) for constructor Hehe: (implicit evidence$1: Playground.TypeClass[T]): Playground.Hehe[T]
			this(tag.runtimeClass.asInstanceOf[Class[T]], tpe)
```

Fortunately, changing from the type class notation to explicit implicit parameter list fixes the issue. I double checked if I wasn't going crazy, but I do have plenty of classes with constructors using ~view~context bounds.

Dotty is more misterioso:
```
➜  snips ~/scala3-3.0.0-M1/bin/scalac -d /tmp t12233.scala
-- Error: t12233.scala:7:54 ----------------------------------------------------
7 |    this(tag.runtimeClass.asInstanceOf[Class[T]], tpe)
  |                                                      ^
  |ambiguous implicit arguments: both value hehe and value evidence$1 match type TypeClass[T] of an implicit parameter of constructor Hehe in class Hehe
1 error found
```
Recall that context bound evidence is prepended to the implicit parameter section:
```
➜  snips cat t12233c.scala

trait TypeClass[T]
class Hehe[T: TypeClass](i: Int, j: Int) {
  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
}
```
whence the power of three:
```
➜  snips scalac -Vprint:typer t12233c.scala
t12233c.scala:4: error: too many arguments (found 3, expected 1) for constructor Hehe: (implicit evidence$1: TypeClass[T]): Hehe[T]
  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
                                                          ^
[[syntax trees at end of                     typer]] // t12233c.scala
package <empty> {
  abstract trait TypeClass[T] extends scala.AnyRef;
  class Hehe[T] extends scala.AnyRef {
    <paramaccessor> private[this] val i: Int = _;
    <paramaccessor> private[this] val j: Int = _;
    implicit <synthetic> <paramaccessor> private[this] val evidence$1: TypeClass[T] = _;
    def <init>(i: Int, j: Int)(implicit evidence$1: TypeClass[T]): Hehe[T] = {
      Hehe.super.<init>();
      ()
    };
    def <init>(i: Int)(implicit evidence$2: TypeClass[T], hehe: TypeClass[T], j: Int): Hehe[T] = {
      Hehe.this.<init>(i, j)(evidence$2, hehe, j);
      ()
    }
  }
}

1 error
```
For some reason, it uses the wrong parameter list.

--- a/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
+++ b/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
@@ -2863,8 +2863,7 @@ self =>
           t = Apply(t, argumentExprs())
           newLineOptWhenFollowedBy(LBRACE)
         }
-        if (classContextBounds.isEmpty) t
-        else Apply(t, vparamss.last.map(vp => Ident(vp.name)))
+        t
       }
 
     /** {{{

--- a/test/files/neg/t12233.check
+++ b/test/files/neg/t12233.check
@@ -1,0 +1,7 @@
+t12233.scala:4: error: ambiguous implicit values:
+ both value hehe of type TypeClass[T]
+ and value evidence$2 of type TypeClass[T]
+ match expected type TypeClass[T]
+  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
+                                                          ^
+1 error

--- a/test/files/neg/t12233.scala
+++ b/test/files/neg/t12233.scala
@@ -1,0 +1,20 @@
+
+trait TypeClass[T]
+class Hehe[T: TypeClass](i: Int, j: Int) {
+  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
+}
+
+/* was
+t12233.scala:4: error: too many arguments (found 3, expected 1) for constructor Hehe: (implicit evidence$1: TypeClass[T]): Hehe[T]
+  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
+                                                          ^
+1 error
+ * now
+t12233.scala:4: error: ambiguous implicit values:
+ both value hehe of type TypeClass[T]
+ and value evidence$2 of type TypeClass[T]
+ match expected type TypeClass[T]
+  def this(i: Int)(implicit hehe: TypeClass[T], j: Int) = this(i, j)
+                                                          ^
+1 error
+ */

--- a/test/files/pos/t12233.scala
+++ b/test/files/pos/t12233.scala
@@ -1,0 +1,12 @@
+
+trait TypeClass[T]
+class Hehe[T: TypeClass](i: Int, j: Int) {
+  def this(i: Int)(implicit j: Int) = this(i, j)
+}
+
+/* was
+test/files/pos/t12233.scala:4: error: too many arguments (found 2, expected 1) for constructor Hehe: (implicit evidence$1: TypeClass[T]): Hehe[T]
+  def this(i: Int)(implicit j: Int) = this(i, j)
+                                      ^
+1 error
+ */

